src/componentWithName.js

```diff
@@ -37,17 +65,27 @@ function hasName(name, prop, propName, componentName, ...rest) {
   return null;
 }
 
-export default function componentWithName(name) {
+export default function componentWithName(name, ...namesOfHOCsToStrip) {
```

Comment:
🤔 my only concern is that this notation isn't very intuitive.

```jsx
const propTypes = {
  children: componentWithName(/^(?!Flow|Radio).*Button$/, 'withStyles', 'withTracking', 'withHOC'),
};
```

Is this a common notation?

How about, 
```jsx
const propTypes = {
  children: componentWithName(/^(?!Flow|Radio).*Button$/, {
    allowHOCs: ['withStyles', 'withTracking', 'withHOC']
  }),
};
```

Would this be too explicit? WDYT?

Reply:
It’s the current pattern in our internal HOC-taking propType, but i like your idea better - it’s also more extensible in the future.